src/commands/function-restore.json

```diff
@@ -4,7 +4,7 @@
         "complexity": "PATCH__TBD__32__",
         "group": "scripting",
         "since": "7.0.0",
-        "arity": 3,
+        "arity": -3,
```

Comment:
is the APPEND/FLUSH/REPLACE an optional or mandatory arg?
if it's optional, it must be last, if it's mandatory, then just put 4 here

Reply:
Its optional, but I prefer not to put it last so the blob could be used with `-x` argument on `redis-cli`. We have this pattern on `FUNCTION CREATE` where `REPLACE` is optional and comes before the code. If you do not like it here maybe you do not like it there either?

Reply:
ohh, i see the code now.

so we're reserving the last argument to the code payload, and any extra arguments in between RESTORE and the payload could be modifiers (more can be added in the future).
kinda like https://redis.io/commands/brpop (last argument is always TIMEOUT, and any args in between are keys).
i think it's a bit awkward.
also, in normal RESTORE the payload is not necessarily the last argument.
maybe we don't absolutely have to have it last?
i.e. you wanted the user to be able to put the function code in a file and use `-x`, but maybe we're willing to waive that?

Reply:
I thought it make sense to build the commands such that we can use the `-x`, I believe `RESTORE` should have also built this way. Another option is to extend `redis-cli` with `--restore-functions` option (like we have for --eval).

Reply:
well, you know what.. i agree with you.
commands with huge binary payloads should be designed so that they have that as a last argument...

or maybe another alternative is to enhance the `-x` feature.
it could be `-x1` and `-x-1` (i.e. to denote injecting the argument at a certain offset from the start or the end).

or it could be `-x<PLACEHOLDER>`
like `redis-cli -xPAYLOAD function restore PAYLOAD << payload.dump`

Reply:
thinking about this more, if it was possible to change RESTORE in a way that the payload is always the last argument, i'd go for it.

but since it's not possible, and actually that -x feature is just an odd feature of redis-cli, maybe we should indeed improve redis-cli to be able to inject stdin at any position like my placeholder idea above, and not worry about the position of that argument in the comment (we don't need to design command according to limitations of redis-cli)

Reply:
Convinced,  will change FLUSH|APPEND|REPLACE to be last argument.

Reply:
Fixed and updated top comment.